tests/containers/list_test.cpp

```diff
@@ -1,56 +1,100 @@
-#include "gtest/gtest.h"
-#include "containers/List.h"
-
-cuhksz::List<int> testList;
-
-TEST(listTest, push) {
-    testList.push(1);
-    testList.push(2);
-    testList.push(3);
-    EXPECT_EQ(testList.size(), 3);
-}
-
-TEST(listTest, push_front) {
-    testList.push_front(5);
-    EXPECT_EQ(testList.first(), 5);
-}
-
-TEST(listTest, insert) {
-    testList.insert(0, 1);
-    EXPECT_EQ(testList[0], 1);
-    EXPECT_EQ(testList.size(), 5);
-
-    testList.insert(2, 4);
-    EXPECT_EQ(testList[2], 4);
-    EXPECT_EQ(testList.size(), 6);
-}
-
-TEST(listTest, erase) {
-    testList.erase(2);
-    EXPECT_EQ(testList.size(), 5);
-}
-
-TEST(listTest, pop) {
-    EXPECT_EQ(testList.pop(), 3);
-}
-
-TEST(listTest, clear) {
-    testList.clear();
-    EXPECT_TRUE(testList.isEmpty());
-}
-
-TEST(listDeathTest, outrangeDeath) {
-    EXPECT_DEATH(testList[8] = 1, "");
-}
-
-TEST(listDeathTest, emptyPopDeath) {
-    EXPECT_DEATH(testList.pop(), "");
-}
-
-// TEST(forDeathTest, rangeForLoop) {
-// 	EXPECT_EXIT(
-// 		for (auto x:testList) {continue;},
-// 		testing::ExitedWithCode(0),
-// 		""
-// 		);
-// }
+#include "gtest/gtest.h"
+#include "containers/List.h"
+#include <list>
+
+cuhksz::List<int> testList;
+std::list<int> testStlList = {1, 2, 3, 4};
+
+TEST(listTest, initialize) {
+    cuhksz::List<int> initList1(5, 1);
+    cuhksz::List<int> initList2 = {1, 2, 1};
+    cuhksz::List<int> initList3(testStlList);
+}
+
+TEST(listTest, typeConvert) {
+    auto list1 = testList.toStlList();
+    std::list<int> list2 = testList;
+    EXPECT_EQ(typeid(list1), typeid(std::list<int>));
+    EXPECT_EQ(typeid(list2), typeid(std::list<int>));

```

Comment:
Do not check this, should check equal in value. As the typeid is determined in the compile time